src/Authentication/Authenticators/Session.php

```diff
@@ -281,51 +315,63 @@ private function checkUserState(): void
         }
 
         /** @var int|string|null $userId */
-        $userId = session(setting('Auth.sessionConfig')['field']);
+        $userId = $this->getSessionUser('id');
 
+        // Has User Info in Session.
         if ($userId !== null) {
             $this->user = $this->provider->findById($userId);
 
-            $identities = $this->userIdentityModel->getIdentitiesByTypes(
-                $this->user->getAuthId(),
-                $this->getActionTypes()
-            );
-
-            // If we will have more than one identity, we need to change the logic blow.
-            assert(
-                count($identities) < 2,
-                'More than one identity for actions. user_id: ' . $userId
-            );
-
-            // Having an action?
-            foreach ($identities as $identity) {
-                $actionClass = setting('Auth.actions')[$identity->name];
-
-                if ($actionClass) {
-                    $this->userState = self::STATE_PENDING;
+            // If having `auth_action`, it is pending.
+            if ($this->getSessionUser('auth_action')) {
+                $this->userState = self::STATE_PENDING;
 
-                    session()->set('auth_action', $actionClass);
-                    $this->pendingMessage = $identity->extra;
-
-                    return;
-                }
+                return;
             }
 
             $this->userState = self::STATE_LOGGED_IN;
 
             return;
         }
 
+        // No User Info in Session.
         // Check remember-me token.
         if (setting('Auth.sessionConfig')['allowRemembering']) {
-            $this->checkRememberMe();
+            if ($this->checkRememberMe()) {
+                $this->setAuthAction();
+            }
 
             return;
         }
 
         $this->userState = self::STATE_ANONYMOUS;
     }
 
+    /**
+     * Gets identities for action from database, and set session.
+     */
+    private function setAuthAction()
+    {
+        // Get identities for action
+        $identities = $this->userIdentityModel->getIdentitiesByTypes(
+            $this->user->getAuthId(),
```

Comment:
```suggestion
            $this->user->id,
```

Weren't we going to get rid of the `getAuthX` methods?

Reply:
The param was changed to User. Rebased and fixed.

`getAuthId()` is still used in a lot of places.
We certainly do not need those methods now and I would like to remove them, 
but the task is not high on my priority list.